api/app/signals/apps/email_integrations/signal_receivers.py

```diff
@@ -3,9 +3,40 @@
 from django.dispatch import receiver
 
 from signals.apps.email_integrations import tasks
-from signals.apps.signals.managers import create_initial, update_status
+from signals.apps.signals.managers import (
+    create_initial,
+    update_signal_departments,
+    update_status,
+    update_user_assignment
+)
 
 
 @receiver((create_initial, update_status), dispatch_uid='core_email_integrations_update_status')
 def update_status_handler(sender, signal_obj, *args, **kwargs):
     tasks.send_mail_reporter.delay(pk=signal_obj.pk)
+
+
+@receiver(update_signal_departments, dispatch_uid='core_email_integrations_update_signal_departments')
+def update_signal_departments_handler(sender, signal_obj, signal_departments, prev_signal_departments, *args, **kwargs):
+    if signal_departments and prev_signal_departments:
```

Comment:
Please move these specific checks if the email action can be triggered to a specific rule `AssignedRule`.